packages/neon_framework/lib/src/widgets/dialog.dart

```diff
@@ -523,3 +555,395 @@ class NeonUnifiedPushDialog extends StatelessWidget {
+          isDestructiveAction: true,
+          onPressed: () {
+            bloc.clearMessage();
+          },
+          child: Text(NeonLocalizations.of(context).userStatusActionClear),
+        ),
+        NeonDialogAction(
+          isDefaultAction: true,
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+          child: Text(NeonLocalizations.of(context).actionClose),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildOnlineStatuses() {
+    final onlineStatuses = {
+      'online': (
+        ['online'],
+        NeonLocalizations.of(context).userStatusOnline,
+        null,
+      ),
+      'dnd': (
+        ['dnd'],
+        NeonLocalizations.of(context).userStatusDoNotDisturb,
+        NeonLocalizations.of(context).userStatusDoNotDisturbDescription
+      ),
+      'away': (
+        ['away'],
+        NeonLocalizations.of(context).userStatusAway,
+        null,
+      ),
+      'invisible': (
+        ['invisible', 'offline'],
+        NeonLocalizations.of(context).userStatusInvisible,
+        NeonLocalizations.of(context).userStatusInvisibleDescription
+      ),
+    };
+
+    return ResultBuilder.behaviorSubject(
+      subject: bloc.status,
+      builder: (context, result) {
+        final onlineStatusTiles = onlineStatuses.entries.map(
+          (onlineStatus) {
+            final (types, title, subtitle) = onlineStatus.value;
+            return Card(
+              shape: types.contains(result.data?.status) ? selectedBorder : null,
+              child: InkWell(
+                borderRadius: const BorderRadius.all(Radius.circular(12)),
+                onTap: () {
+                  bloc.setStatusType(onlineStatus.key);
+                },
+                child: Center(
+                  child: ListTile(
+                    leading: NeonServerIcon(icon: 'user-status-${onlineStatus.key}'),
+                    title: Text(title),
+                    subtitle: subtitle != null ? Text(subtitle) : null,
+                  ),
+                ),
+              ),
+            );
+          },
+        ).toList();
+
+        return Column(
+          children: [
+            NeonError(
+              result.error,
+              onRetry: bloc.refresh,
+            ),
+            NeonLinearProgressIndicator(
+              visible: result.isLoading,
+            ),
+            buildGrid(NeonDialogTheme.of(context).constraints, onlineStatusTiles),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildPredefinedStatuses() => ResultBuilder.behaviorSubject(
+        subject: bloc.predefinedStatuses,
+        builder: (context, predefinedStatusesResult) => ResultBuilder.behaviorSubject(
+          subject: bloc.status,
+          builder: (context, statusResult) {
+            final selectedPredefinedStatus = statusResult.hasData && statusResult.requireData is user_status.Private
+                ? (statusResult.requireData as user_status.Private).messageId
+                : null;
```

Comment:
```suggestion
            final statusResultData = statusResult.data;
            String? selectedPredefinedStatus;
            if (statusResultData is user_status.Private) {
              selectedPredefinedStatus = statusResultData.messageId;
            }
```
or
```suggestion
            final statusResultData = statusResult.data;
            final selectedPredefinedStatus = statusResultData is user_status.Private? 
              selectedPredefinedStatus = statusResultData.messageId: null;
```


Local variables can be promoted by typechecks. I think this is cleaner.